api/core/model_runtime/model_providers/bedrock/llm/llm.py

```diff
@@ -75,8 +77,86 @@ def _invoke(self, model: str, credentials: dict,
         # invoke anthropic models via anthropic official SDK
         if "anthropic" in model:
             return self._generate_anthropic(model, credentials, prompt_messages, model_parameters, stop, stream, user)
+        # invoke Cohere models via boto3 client
+        if "cohere.command-r" in model:
+            return self._generate_cohere_chat(model, credentials, prompt_messages, model_parameters, stop, stream, user, tools)
         # invoke other models via boto3 client
         return self._generate(model, credentials, prompt_messages, model_parameters, stop, stream, user)
+    
+    def _generate_cohere_chat(
```

Comment:
i don't think an extra method is needed. can the cohere-specific logic be added under _handle_generate_stream_response?

Reply:
It's not about the response, it's about the different format of the data passed to bedrock.

refer to [aws becrock docs]https://docs.aws.amazon.com/zh_cn/bedrock/latest/userguide/model-parameters-cohere-command-r-plus.html .